line_profiler/_line_profiler.pyx

```diff
@@ -128,9 +145,39 @@ cdef class LineProfiler:
             import warnings
             warnings.warn("Could not extract a code object for the object %r" % (func,))
             return
-        if code not in self.code_map:
-            self.code_map[code] = {}
-            self.functions.append(func)
+
+        if code.co_code in self.dupes_map:
+            self.dupes_map[code.co_code] += [code]
+            # code hash already exists, so there must be a duplicate function. add no-op
```

Comment:
Can you give an example of a case that would trigger this? The old code doesn't seem to have handling for this case, so I don't understand why we are adding the no-op instruction.

Reply:
Sure!
test.py:
```py
import test2
@profile
def run():
    x = [1 for _ in range(1000)]
    return x

run()
```
test2.py:
```py

@profile
def run():
    x = [1 for _ in range(1000)]
    return x

run()
```
Without adding the no-op, line_profiler v4 is unable to distinguish between the two functions. In the old version, I believe function code objects are identified by their memory address, whereas now they're identified by their code and the line's number. In the above case, the code and line numbers are the same, so adding the no-op makes the code hashes different, and therefore the functions distinguishable.

Reply:
(This is actually the code I'm most proud of in this PR, I find this to be an elegant solution for the issue of distinguishing between functions without being able to access their memory addresses directly from C++)